commit 4b246b2bbd1d5a77035bb990d6097b7337c34bbb
Author: Adhemerval Zanella <adhemerval.zanella@linaro.org>
Date:   Thu Jun 30 09:08:31 2022 -0300

    linux: Fix mq_timereceive check for 32 bit fallback code (BZ 29304)
    
    On  success,  mq_receive() and mq_timedreceive() return the number of
    bytes in the received message, so it requires to check if the value
    is larger than 0.
    
    Checked on i686-linux-gnu.
    
    (cherry picked from commit 71d87d85bf54f6522813aec97c19bdd24997341e)

diff --git a/sysdeps/unix/sysv/linux/mq_timedreceive.c b/sysdeps/unix/sysv/linux/mq_timedreceive.c
index 7f3a112d7f2cbbe7..1fc98752e7d6d506 100644
--- a/sysdeps/unix/sysv/linux/mq_timedreceive.c
+++ b/sysdeps/unix/sysv/linux/mq_timedreceive.c
@@ -41,7 +41,7 @@ ___mq_timedreceive_time64 (mqd_t mqdes, char *__restrict msg_ptr, size_t msg_len
     {
       int r = SYSCALL_CANCEL (mq_timedreceive_time64, mqdes, msg_ptr, msg_len,
 			      msg_prio, abs_timeout);
-      if (r == 0 || errno != ENOSYS)
+      if (r >= 0 || errno != ENOSYS)
 	return r;
       __set_errno (EOVERFLOW);
       return -1;
